### PR TITLE
hw-007-solution

### PR DESCRIPTION
**Pull request by SviatoslavExpert**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -5,15 +5,46 @@
...
+                }
+            }
+        }
+        return result;
     }
 
-    public int getMin(int[][] matrix) {
-        return 0;
+    protected int getMin(int[][] matrix) {
+        int minimum = Integer.MAX_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why we use Integer.MAX_VALUE?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -5,15 +5,46 @@
...
+                }
+            }
+        }
+        return minimum;
     }
 
-    public int getMax(int[][] matrix) {
-        return 0;
+    protected int getMax(int[][] matrix) {
+        int maximum = Integer.MIN_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: the same?

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/MatrixOperation.java]: @SviatoslavExpert lets use first element of matrix as we have discussed on the lesson

Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed.  Thank you.

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed.  Thank you.

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -5,15 +5,46 @@
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
-    public int[] getDiagonal(int[][] matrix) {
-        return null;
+    protected int[] getDiagonal(int[][] matrix) {
+        int row = matrix.length;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to simplify this method
we work with square matrix in this task, so we can use one for loop here

Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Here is what it says in the technical description which goes with the code:  
Реалізуйте методи для знаходження головної діагоналі **прямокутної** матриці,
максимального та мінімального елементів.
So, it is not square matrix, it is a **rectangular** matrix.  Let's not change the task on the way.

**Pull request closed without merging**